pkg/controllermanager/controller/controllerregistration/seed/add.go

```diff
@@ -360,3 +360,10 @@ func (r *Reconciler) MapControllerDeploymentToAllSeeds(ctx context.Context, log
 
 	return nil
 }
+
+func shootNetworkingTypeHasChanged(old, new *gardencorev1beta1.Networking) bool {
+	if old == nil || new == nil {
+		return false
+	}
```

Comment:
This cannot happen since we actually return if the type has changed. For Shoot with workers neither old or nil can be nil, and for workerless neither of them has type. 
But from controller POV, maybe we can add it, yes.

Reply:
OK, yes, let's consider the controller's PoV here.

Reply:
Addressed, PTAL!